### DISCUSSION TITLE
getting ISO 8601 formatted date string back

### DISCUSSION THREAD

**Discussion by astalea** [matheo/angular#13] [Q&A]
I the following code:

```
<mat-label>Choose a datetime</mat-label>
<input matInput [(ngModel)]="myDate" [matDatepicker]="picker" />
<mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
<mat-datepicker type="datetime" [twelveHour]="false" #picker>
  <mat-datepicker-actions>
    <button mat-button matDatepickerCancel>Cancel</button>
    <button mat-raised-button matDatepickerApply (click)="updateFromApply(myDate)">Apply</button>
  </mat-datepicker-actions>
</mat-datepicker>
</mat-form-field>
```

myDate is an ISO 8601 formatted date string and initializes mat-datepicker correctly. However after changing date/time vi UI I am getting a localized date time string back. 

How to get an ISO 8601 formatted date string instead?

**Comment by matheo** [comment]
@astalea the datepicker returns a `Date` object that is converted to `string` when you console.log it
try to post-process it in your `updateFromApply` method, like:
```typescrript
updateFromApply(): void {
  console.log(this.myDate.toISOString());
  ...
}
```
ideally you should build a custom `DateAdapter` to output the format you want directly to `formControlName`/`ngModel` but that's kinda a lot of work for this simple case, so it's also possible to update your value (or patch your form) if the value you have is `Date` (or not `string`):
```
myDate: any;

updateFromApply(): void {
  this.myDate = typeof this.myDate === 'string' ? this.myDate : this.myDate.toISOString();
}
```
